nat/mapping/port_mapping.go

```diff
@@ -73,14 +83,19 @@ func mapPort(m portmap.Interface, c chan struct{}, protocol string, extPort, int
 	}
 }
 
-func addMapping(m portmap.Interface, protocol string, extPort, intPort int, name string) {
-	if err := m.AddMapping(protocol, extPort, intPort, name, mapTimeout); err != nil {
-		log.Debugf("%s, Couldn't add port mapping for port %d: %v, retrying with permanent lease", logPrefix, extPort, err)
-		if err := m.AddMapping(protocol, extPort, intPort, name, 0); err != nil {
-			// some gateways support only permanent leases
-			log.Debugf("%s Couldn't add port mapping for port %d: %v", logPrefix, extPort, err)
-			return
-		}
+func addMapping(m portmap.Interface, protocol string, extPort, intPort int, name string) error {
+	err := m.AddMapping(protocol, extPort, intPort, name, mapTimeout)
```

Comment:
I personally feel the old version of the method was way more readable, the `err == nil` check feels super odd in what is usually an `error != nil` world.

Reply:
I personally found it very hard to understand all the cases in the previous example - but we're all different and it might be just me.
My motivation was to remove nesting by returning early - i.e. if we had to more more edge-cases, we would need more nested `if`s in first approach, and in my approach - only single method-level `return` would be added.

http://www.itamarweiss.com/personal/2018/02/28/return-early-pattern.html
https://softwareengineering.stackexchange.com/questions/18454/should-i-return-from-a-function-early-or-use-an-if-statement

Do others find this more difficult to understand as well?

Reply:
I realise that multiple nested levels are a bit fishy but in this case we're not nesting that much.